### DISCUSSION TITLE
Feature for rearranging folders and files in Sidebar/LeftDock/Project Structure in any order.

### DISCUSSION THREAD

**Discussion by Ilham-Habibullin** [zed-industries/zed#8510] [Feature Requests]
Currently in Zed and almost every other code editors/IDEs order of folders and files is constrained to be alphabetic. In Xcode though it is allowed to rearrange(using drag&drop) files and folders and place them in any order user wants. Very nice feature that allows to keep files/folders in accordance with their meaning instead of their names. 

How about implementing same thing in Zed?

**Comment by Moshyfawn** [comment]
Do you think #4851 could be an umbrella feature request to cover your use case? Or are you talking about a completely custom, user-defined order?

**Reply by Moshyfawn** [reply]
It does seem like [they're after rearranging files completely out of order](https://github.com/zed-industries/zed/issues/4851#issuecomment-1908635710), beyond any sorting algorithms. I'd advise you to +👍 the issue and possibly expand on your use case in the comment in the issue thread.

**Reply by Ilham-Habibullin** [reply]
I think what he is proposing [here](https://github.com/zed-industries/zed/issues/4851#issuecomment-1908635710) is something like a custom rules for sorting e.g. alphabetical or numerical or folders-first-files-last or by-creation-time, while what i mean is manual rearranging like in this GIF 

![ezgif-4-37ca8f3fe0](https://github.com/zed-industries/zed/assets/18722721/a2b9021e-d3f1-4e69-9924-d35ea7041a01)

**Reply by Ilham-Habibullin** [reply]
@Moshyfawn i left a [question](https://github.com/zed-industries/zed/issues/4851#issuecomment-1969152048) there

**Reply by Moshyfawn** [reply]
[This particular comment](https://github.com/zed-industries/zed/issues/4851#issuecomment-1908635710 ) makes me think you're both after out-of-order reordering; note how he's got the selected file in the middle of an ordered list.

Also, thanks for posting in that thread 👍

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
